realm/src/androidTest/java/io/realm/DynamicRealmObjectTest.java

```diff
@@ -0,0 +1,334 @@
+package io.realm;/*
+ * Copyright 2014 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.test.AndroidTestCase;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
+import io.realm.dynamic.DynamicRealmList;
+import io.realm.dynamic.DynamicRealmObject;
+import io.realm.entities.AllJavaTypes;
+
+import static io.realm.internal.test.ExtraTests.assertArrayEquals;
+
+public class DynamicRealmObjectTest extends AndroidTestCase {
+
+    private Realm realm;
+    private DynamicRealmObject dObj;
+
+    @Override
+    protected void setUp() throws Exception {
+        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext()).schema(AllJavaTypes.class).build();
+        Realm.deleteRealm(realmConfig);
+        realm = Realm.getInstance(realmConfig);
+        realm.beginTransaction();
+        AllJavaTypes obj = realm.createObject(AllJavaTypes.class);
```

Comment:
AllJavaTypes should have it's fields renamed replacing "Column" with "Field" - but in a separate PR.